src/main/java/org/testeditor/web/xtext/index/serialization/EObjectDescriptionDeserializer.xtend

```diff
@@ -0,0 +1,50 @@
+package org.testeditor.web.xtext.index.serialization
+
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.core.TreeNode
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.node.TextNode
+import java.io.IOException
+import java.util.regex.Pattern
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.EClass
+import org.eclipse.emf.ecore.EPackage
+import org.eclipse.xtext.naming.QualifiedName
+import org.eclipse.xtext.resource.IEObjectDescription
+import org.eclipse.xtext.resource.persistence.SerializableEObjectDescription
+
+import static org.testeditor.web.xtext.index.serialization.EObjectDescriptionSerialization.EOBJECT_URI__FIELD_NAME
+import static org.testeditor.web.xtext.index.serialization.EObjectDescriptionSerialization.QUALIFIED_NAME__DELIMITER
+import static org.testeditor.web.xtext.index.serialization.EObjectDescriptionSerialization.QUALIFIED_NAME__FIELD_NAME
+import static org.testeditor.web.xtext.index.serialization.EObjectDescriptionSerialization.URI__FIELD_NAME
+
+class EObjectDescriptionDeserializer extends StdDeserializer<IEObjectDescription> {
+
+	new(Class<?> vc) {
```

Comment:
Is this ctor necessary?

Reply:
Why not define ctor without parameters, passing super IEObjectDescription?

Reply:
I need a public constructor, but you're right, the argument is completely redundant.